assets/src/components/propertiesPanel/minischedule.tsx

```diff
@@ -107,7 +131,18 @@ const Piece = ({ piece, view }: { piece: Piece; view: "run" | "block" }) => (
             : index === piece.trips.length - 1
             ? "last"
             : "middle"
-        return <Trip trip={trip} sequence={sequence} key={trip.id} />
+        return (
+          <React.Fragment key={trip.id}>
+            <Trip trip={trip} sequence={sequence} />
+            {view === "run" ? (
+              <Layover
+                currentTrip={trip}
+                nextTrip={piece.trips[index + 1]}
+                key={`layover-${trip.endTime}`}
+              />
+            ) : null}
+          </React.Fragment>
```

Comment:
Laying over vehicles are assigned to their upcoming trip in the realtime data. I think that the active trips ticket might be easier if the layover component here was drawn before the trip with the same id, instead of after the previous trip, so the two components in the fragment are checking for the same `trip.id`, instead of one checking for `trip.id` and one for `piece.trips[index+1]?.id`.

```jsx
<>
  <Layover
    previousTrip={piece.trips[index-1]}
    nextTrip={trip}
  />
  <Trip/>
</>
```

(But you're the one doing that work so you know better than me if this would help.)

Reply:
@skyqrose hmm yes I think that might be a bit more elegant, I'll try it

Reply:
@skyqrose actually I'm looking at the active-trip-row card and it says:

> If vehicle is laying over, highlight the smaller "Layover [duration in min]" row *below* the completed trip as well.

[emphasis mine], so I think the current implementation is preferable after all.